### PR TITLE
create primary profile on account creation

### LINKED ISSUE #63: add process to set account holder profile
the account holder must have the primary profile, and cannot be deleted.

### PR DESCRIPTION
**Pull request by joshwingreene**
Closes #63

### CODE REVIEW

**Comment by fredbelotte** [comment]
where are we with this?

**Comment by joshwingreene** [comment]
@fredbelotte This PR is ready to be reviewed. I forgot to add you as a reviewer.

File: aspnet/RVTR.Account.Context/AccountContext.cs [hunk 1]
```diff
@@ -27,25 +27,29 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
         new AccountModel
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Context/AccountContext.cs:27]: let's reduce to 1 complete account

File: aspnet/RVTR.Account.Domain/Models/AccountModel.cs [hunk 2]
```diff
@@ -14,30 +14,42 @@ public class AccountModel : AEntity, IValidatableObject
...
 
-    [Required(ErrorMessage = "Name required")]
-    [MaxLength(50, ErrorMessage = "Name must be fewer than 50 characters.")]
-    [RegularExpression(@"^[A-Z]+[a-zA-Z""'\s-]*$", ErrorMessage = "Name must start with a capital letter and only use letters.")]
-    public string Name { get; set; }
-
-    public IEnumerable<PaymentModel> Payments { get; set; } = new List<PaymentModel>();
-
-    public IEnumerable<ProfileModel> Profiles { get; set; } = new List<ProfileModel>();
+    [Required(ErrorMessage = "First name required")]
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Domain/Models/AccountModel.cs:17]: remove the capitalization rule on Name, where is LastName

File: aspnet/RVTR.Account.Domain/Models/AccountModel.cs [hunk 3]
```diff
@@ -14,30 +14,42 @@ public class AccountModel : AEntity, IValidatableObject
...
+    [MaxLength(50, ErrorMessage = "First name must be fewer than 50 characters.")]
+    [RegularExpression(@"^[A-Z]+[a-zA-Z""'\s-]*$", ErrorMessage = "First name must start with a capital letter and only use letters.")]
+    public string FirstName { get; set; }
 
+    [Required(ErrorMessage = "Last name required")]
+    [MaxLength(50, ErrorMessage = "Last name must be fewer than 50 characters.")]
+    [RegularExpression(@"^[A-Z]+[a-zA-Z""'\s-]*$", ErrorMessage = "Last name must start with a capital letter and only use letters.")]
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Domain/Models/AccountModel.cs:24]: remove capitalization rule,

File: aspnet/RVTR.Account.Domain/Models/AccountModel.cs [hunk 4]
```diff
@@ -48,9 +60,13 @@ public AccountModel(string name, string email)
     /// <returns></returns>
     public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
     {
-      if (string.IsNullOrEmpty(Name))
+      if (string.IsNullOrEmpty(FirstName))
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Domain/Models/AccountModel.cs:63]: this is handled by the Required attribute

File: aspnet/RVTR.Account.Domain/Models/ProfileModel.cs [hunk 5]
```diff
@@ -9,6 +9,8 @@ namespace RVTR.Account.Domain.Models
   public class ProfileModel : AEntity, IValidatableObject
   {
 
+    public bool IsAccountHolder { get; set; }
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Domain/Models/ProfileModel.cs:12]: is this not handled by the account, why allowing Set here?

File: aspnet/RVTR.Account.Domain/Models/AccountModel.cs [hunk 6]
```diff
@@ -14,30 +14,42 @@ public class AccountModel : AEntity, IValidatableObject
...
-    [RegularExpression(@"^[A-Z]+[a-zA-Z""'\s-]*$", ErrorMessage = "Name must start with a capital letter and only use letters.")]
-    public string Name { get; set; }
-
-    public IEnumerable<PaymentModel> Payments { get; set; } = new List<PaymentModel>();
-
-    public IEnumerable<ProfileModel> Profiles { get; set; } = new List<ProfileModel>();
+    [Required(ErrorMessage = "First name required")]
+    [MaxLength(50, ErrorMessage = "First name must be fewer than 50 characters.")]
+    [RegularExpression(@"^[A-Z]+[a-zA-Z""'\s-]*$", ErrorMessage = "First name must start with a capital letter and only use letters.")]
+    public string FirstName { get; set; }
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Domain/Models/AccountModel.cs:20]: doesn't Name part of the Profile already?

File: aspnet/RVTR.Account.Testing/Tests/RepositoryTest.cs [hunk 7]
```diff
@@ -130,18 +130,18 @@ public async void Test_Repository_SelectAsync_ById()
...
-        var account = await ctx.Accounts.FirstAsync();
-
-        account.Name = "name";
-        accounts.Update(account);
-
-        var result = ctx.Accounts.Find(account.EntityId);
-        Assert.Equal(account.Name, result.Name);
-        Assert.Equal(EntityState.Modified, ctx.Entry(result).State);
-      }
+      // using (var ctx = new AccountContext(Options))
```
Comment (fredbelotte) [inline aspnet/RVTR.Account.Testing/Tests/RepositoryTest.cs:133]: remove commented code

**Review by fredbelotte — CHANGES_REQUESTED** [review]

**Comment by joshwingreene** [comment]
@fredbelotte All of the issues that you pointed out have been fixed and this PR is ready for review again.

**Review by fredbelotte — APPROVED** [review]
add new issue to refactor profileModel and accountModel
add new issue to update the unit tests

**Pull request merged**